x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -240,10 +240,15 @@ public static boolean canPushToSource(Expression exp) {
             } else if (exp instanceof Not not) {
                 return canPushToSource(not.field());
             }
-            return false;
+            for (Attribute a : exp.references()) {
```

Comment:
I'm afraid this is too generic.
It is problematic with boolean expressions that we are not considering in the previous IF, eg. functions that return a boolean

```
from employees | where to_boolean(to_string(emp_no))
```

```
{
    "error": {
        "root_cause": [
            {
                "type": "esql_illegal_argument_exception",
                "reason": "Expected a FieldAttribute or MetadataAttribute but received [to_boolean(to_string(emp_no))]"
            }
        ],
        "type": "esql_illegal_argument_exception",
        "reason": "Expected a FieldAttribute or MetadataAttribute but received [to_boolean(to_string(emp_no))]"
    },
    "status": 500
}
```

Reply:
Fixed.